server/repository/src/database_settings.rs

```diff
@@ -102,21 +102,39 @@ impl diesel::r2d2::CustomizeConnection<SqliteConnection, diesel::r2d2::Error>
 pub fn get_storage_connection_manager(settings: &DatabaseSettings) -> StorageConnectionManager {
     let connection_manager =
         ConnectionManager::<DBBackendConnection>::new(&settings.connection_string());
-    let pool = Pool::new(connection_manager).expect("Failed to connect to database");
+    let result = Pool::new(connection_manager);
+    let pool = match result {
+        Ok(pool) => pool,
+        Err(err) => panic!(
```

Comment:
hmm, `https://doc.rust-lang.org/std/result/enum.Result.html#method.expect` should return content of error, what is the difference in output with this change ?

Reply:
The only difference now would be stating sqlite or postgres but agreed, it's not really that different.
A log message like "Starting with SQLite" or "Starting with Postgres"
I didn't realise that expect already returns the full error along with the message. 

Had this ticket wrong from the start :)

Reply:
Ok, yeah as per earlier comment above, that error is the result of using anyhow, when those errors are unwrapped we only get the derived description:

https://github.com/openmsupply/open-msupply/blob/800fe47cafaca0bce7077d9a02ec071b25461a84/server/server/src/sync/central_data_synchroniser.rs#L29

Reply:
I think you can include source with `#[error("Failed to connect to DB {source?}")]`

as per https://docs.rs/anyhow/latest/anyhow/ `#[error("Invalid header (expected {expected:?}, got {found:?})")]`

Reply:
i personally haven't warmed up to generic error crates yet, I would have preferred manually nesting errors in enums

Reply:
I just tried this locally.

```
pub fn return_error() -> Result<ServiceContext, RepositoryError> {
    return Err(RepositoryError::DBError {
        msg: "TEST ERROR".to_string(),
        extra: "TEST ERROR".to_string(),
    });
}
```

If I do  `let _test = return_error()?;`
I get
`[2022-06-07T23:49:56Z ERROR server] Failed to perform the initial sync: "TEST ERROR" ("TEST ERROR")`

With 
`let _test = return_error().map_err(CentralSyncError::from_database_error)?;`
I get 
`[2022-06-07T23:53:19Z ERROR server] Failed to perform the initial sync: Failed to connect to DB`

With 
` #[error("Failed to connect to DB - {source:?}")]`
I get 
`[2022-06-07T23:58:14Z ERROR server] Failed to perform the initial sync: Failed to connect to DB - DBError { msg: "TEST ERROR", extra: "TEST ERROR" }`

Last one is probably the best option for now?